### ISSUE TITLE
mach_vm_map failed errors on OS X Mavericks

### ISSUE THREAD

**Issue by FooBarWidget** [phusion/passenger#1193]
_From [honglilai](https://code.google.com/u/honglilai@gmail.com/) on May 23, 2014 14:57:31_

The mach_vm_map errors on OS X Mavericks, like these...

PassengerHelperAgent(49856,0x7fff7686c310) malloc: **\* mach_vm_map(size=1048576) failed (error code=268435459)
PassengerHelperAgent(49856,0x7fff7686c310) malloc: **\* error: can't allocate region securely
PassengerHelperAgent(49856,0x7fff7686c310) malloc: **\* set a breakpoint in malloc_error_break to debug

...apparently originate from getHighestFileDescriptor(), when opening in /proc/self/fd. This can be solved by using fork() instead of asyncFork().

_Original issue: http://code.google.com/p/phusion-passenger/issues/detail?id=1094_

**Comment by alxx** [comment]
I still get these errors consistently, though not 100% of the time, on OSX 10.13.4 (High Sierra) and Passenger 5.3.0:

$ passenger start                                                                                                                    ‹ruby-2.4.1›
=============== Phusion Passenger Standalone web server started ===============
PID file: ...tmp/pids/passenger.3000.pid
Log file: ...log/passenger.3000.log
Environment: development
Accessible via: http://0.0.0.0:3000/

You can stop Phusion Passenger Standalone by pressing Ctrl-C.
Problems? Check https://www.phusionpassenger.com/library/admin/standalone/troubleshooting/
===============================================================================
App 69205 output: PassengerAgent(69212,0x7fff9d484380) malloc: *** mach_vm_map(size=8388608) failed (error code=268435459)
App 69205 output: PassengerAgent(69212,0x7fff9d484380) malloc: *** error: can't allocate region securely
App 69205 output: PassengerAgent(69212,0x7fff9d484380) malloc: *** set a breakpoint in malloc_error_break to debug
App 69205 output: PassengerAgent(69214,0x7fff9d484380) malloc: *** mach_vm_map(size=8388608) failed (error code=268435459)

**Comment by daBee** [comment]
I know it's closed, but I found this on a search for the same issue.  Not sure what else I can provide.  File upload on post:

	App 82924 output: PassengerAgent(82927,0x7fff7bd3b000) malloc: *** mach_vm_map(size=1048576) failed (error code=268435459)
	App 82924 output: PassengerAgent(82927,0x7fff7bd3b000) malloc: *** error: can't allocate region securely
	App 82924 output: PassengerAgent(82927,0x7fff7bd3b000) malloc: *** set a breakpoint in malloc_error_break to debug
	App 82924 output: PassengerAgent(82929,0x7fff7bd3b000) malloc: *** mach_vm_map(size=1048576) failed (error code=268435459)
	App 82924 output: PassengerAgent(82929,0x7fff7bd3b000) malloc: *** error: can't allocate region securely
	App 82924 output: PassengerAgent(82929,0x7fff7bd3b000) malloc: *** set a breakpoint in malloc_error_break to debug
	App 82924 output: PassengerAgent(82933,0x7fff7bd3b000) malloc: *** mach_vm_map(size=1048576) fa
	App 82924 output: iled (error code=268435459)
	App 82924 output: PassengerAgent(82933,0x7fff7bd3b000) malloc: *** error: can't allocate region securely
	App 82924 output: PassengerAgent(82933,0x7fff7bd3b000) malloc: *** set a breakpoint in malloc_error_break to debug
	App 82924 output: PassengerAgent(82937,0x7fff7bd3b000) malloc: 
	App 82924 output: *** mach_vm_map(size=8388608) failed (error code=268435459)
	App 82924 output: PassengerAgent(82937,0x7fff7bd3b000) malloc: *** error: can't allocate region securely
	App 82924 output: PassengerAgent(82937,0x7fff7bd3b000) malloc: *** set a breakpoint in malloc_error_break to debug
	App 82924 output: PassengerAgent(82939,0x7fff7bd3b000) malloc: *** mach_vm_map(size=8388608) failed (error code=268435459)
	App 82924 output: PassengerAgent(82939,0x7fff7bd3b000) malloc: *** error: can't allocate region
	App 82924 output:  securely
	App 82924 output: PassengerAgent(82939,0x7fff7bd3b000) malloc: *** set a breakpoint in malloc_error_break to debug
	App 82924 output: PassengerAgent(82980,0x7fff7bd3b000) malloc: *** mach_vm_map(size=1048576) failed (error code=268435459)
	App 82924 output: PassengerAgent(82980,0x7fff7bd3b000) malloc: *** error: can't allocate region securely
	App 82924 output: PassengerAgent(82980,0x7fff7bd3b000) malloc: *** set a breakpoint in malloc_error_break to debug
	App 82924 output: PassengerAgent(82982,0x7fff7bd3b000) malloc: *** mach_vm_map(size=1048576) failed (error code=268435459)
	App 82924 output: PassengerAgent(82982,0x7fff7bd3b000) malloc: *** error: can't allocate region securely
	App 82924 output: PassengerAgent(82982,0x7fff7bd3b000) malloc: *** set a breakpoint in malloc_error_break to debug
	App 82924 output: /Users/rich/.rbenv/versions/2.5.0/lib/ruby/2.5.0/fileutils.rb:90: warning: already initialized constant FileUtils::VERSION
	App 82924 output: /Users/rich/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:92: warning: previous definition of VERSION was here

**Comment by CamJN** [comment]
It's not the malloc messages that are the problem in these cases. Even though they're annoying.

@daBee in your example the real problem is this section:
```
App 82924 output: /Users/rich/.rbenv/versions/2.5.0/lib/ruby/2.5.0/fileutils.rb:90: warning: already initialized constant FileUtils::VERSION
App 82924 output: /Users/rich/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/fileutils-1.1.0/lib/fileutils.rb:92: warning: previous definition of VERSION was here
```
@adammoisa in your case the app launcher seems to be the problem: please provide more details about your config in a new issue.
```
App 76242 output: !> I have control 1.0
[ N 2018-05-31 16:59:18.2573 76227/T5 age/Cor/SecurityUpdateChecker.h:517 ]: Security update check: no update found (next check in 24 hours)
[ W 2018-05-31 16:59:41.2663 76193/T1 age/Wat/WatchdogMain.cpp:470 ]: Some Phusion Passenger agent processes did not exit in time, forcefully shutting down all.
```

**Comment by adammoisa** [comment]
I fully uninstalled and reinstalled and made sure nginx was totally stopped before running `passenger start`, but now, it has a spawn timeout error: 

```
[ E 2018-06-06 08:42:43.6050 84961/Tq age/Cor/App/Implementation.cpp:221 ]: Could not spawn process for application /Users/adammoisa/Desktop/CODE/Projects/leaderboard: A timeout error occurred while spawning an application process.
  Error ID: ee7a9f18
  Error details saved to: /var/folders/vf/l6r3f4lj05vg967w19bl_3480000gn/T//passenger-error-2N7BcI.html

[ E 2018-06-06 08:42:43.6140 84961/Ta age/Cor/Con/CheckoutSession.cpp:276 ]: [Client 2-1] Cannot checkout session because a spawning error occurred. The identifier of the error is ee7a9f18. Please see earlier logs for details about the error.
[ E 2018-06-06 08:42:43.6140 84961/T7 age/Cor/Con/CheckoutSession.cpp:276 ]: [Client 1-1] Cannot checkout session because a spawning error occurred. The identifier of the error is ee7a9f18. Please see earlier logs for details about the error.
```

**Comment by CamJN** [comment]
Yes that's probably still related to the app loader, are you using meteor?

**Comment by CamJN** [comment]
Then your issue was fixed in Passenger 5.3.2.
